addons/mail/static/src/components/discuss/discuss_tests.js

```diff
@@ -5519,6 +5519,303 @@ QUnit.test('moderation: sent message rejected in moderated channel', async funct
+                    record_name: 'General',
+                    res_id: 20,
+                };
+                const notifications = [[
+                    ['my-db', 'res.partner', self.env.session.partner_id],
+                    { type: 'author', message },
+                ]];
+                self.widget.call('bus_service', 'trigger', 'notification', notifications);
+            }
+            return this._super(...arguments);
+        }
+    });
+    await afterNextRender(() =>
+        document.querySelector(`
+            .o_DiscussSidebar_item[data-thread-local-id="mail.thread_mail.channel_20"]
+        `).click()
+    );
+
+    await afterNextRender(() => {
+        document.querySelector('.o_ComposerTextInput').focus();
+        document.execCommand('insertText', false, "Moderated test");
+    });
+    await afterNextRender(() =>
+        document.querySelector('.o_Composer_buttonSend').click()
+    );
+    assert.containsOnce(
+        document.body,
+        '.o_Message',
+        "the pending moderation message should be in the channel"
+    );
+    assert.strictEqual(
+        document.querySelector('.o_Message_content').textContent,
+        "Moderated test",
+        "the pending moderation message should have the correct content"
+    );
+});
+
+QUnit.test('moderation: new pending moderation message posted by someone else should not appear in origin thread', async function (assert) {
+    assert.expect(1);
+
+    const self = this;
+    Object.assign(this.data.initMessaging, {
+        channel_slots: {
+            channel_channel: [{
+                id: 20,
+                channel_type: "channel",
+                name: "general",
+                moderation: true,
+            }],
+        },
+    });
+    await this.start();
+    await afterNextRender(() =>
+        document.querySelector(`
+            .o_DiscussSidebar_item[data-thread-local-id="mail.thread_mail.channel_20"]
+        `).click()
+    );
+
+    // simulate receiving the message
+    const message = {
+        author_id: [10, "Don Diego de la Vega"],
+        body: "hoy !",
+        channel_ids: [], // server do NOT return channel_id of the message if pending moderation
+        moderation_status: 'pending_moderation',
+        id: 1,
+        model: 'mail.channel',
+        record_name: 'General',
+        res_id: 20,
+    };
+    await afterNextRender(() => {
+        const notifications = [[
+            ['my-db', 'res.partner', 10],
+            { type: 'author', message },
+        ]];
+        self.widget.call('bus_service', 'trigger', 'notification', notifications);
+    });
+    assert.containsNone(
+        document.body,
+        '.o_Message',
+        "the pending moderation message should not be in the channel"
+    );
+});
+
+QUnit.test('moderation: new pending moderation message posted by someone else should appear in origin thread and moderation box if I moderate it', async function (assert) {
```

Comment:
Way shorter name for test (this also applies for other tests, but this one is extreme).
Could add comment just above `assert.expect()`